### DISCUSSION TITLE
Select only highest level element selected.

### DISCUSSION THREAD

**Discussion by mcpat1993** [simonwep/viselect#113] [Q&A]
I'm building a tool which needs to behave similar to what window.getSelected() would do except that I want to grab all of the text of any elements in which the selection has partially covered it.
For example this select would get me the string including the text of that full paragraph and not only the highlighted portion:
<img width="942" alt="Screen Shot 2021-05-09 at 17 08 41" src="https://user-images.githubusercontent.com/8905548/117577221-36e86300-b0e9-11eb-80ea-cce0cbabd5f7.png">

I'm using this library to get the elements and then running a textContent over it and I've found that using intersect center works quite well except I was hoping to be able to set some sort of selector such that my selected array only includes the highest level element. As it is now, I'm able to select sibling elements but if they have children for a bolded portion of the paragraph then those elements are also included which I don't want. 

Is there a better way of doing this besides just traversing the children of every node and checking it against all other nodes for overlap?

Thanks in advanced!

Awesome library, Simon!

**Comment by simonwep** [comment]
Hi :) Maybe you could make use of the direct-children selector `>` to limit yourself to only the first level of children? It is definitely interesting to use this library to do this sort of thing, maybe it'd help if could add a JSFiddle / Codesandbox to your question so that we can try it out ourselves - a piece of code is worth a thousand words 😁

Besides that, I guess _until_ you're running into performance issues, traversing the children of every node shouldn't be a problem.
